docs/solana/speclanguage.rst

```diff
@@ -1,8 +1,138 @@
 Certora Verification Language for Rust (CVLR)
 =============================================
 
-CVLR is specification language for formal verification.... 
+CVLR stands for Certora Verification Language for Rust. It is pronounced like "cavalier". 
+CVLR provides the basic primitives for writing verification rules that specify
+pre- and post-conditions for client code. Unlike CVL for Solidity, CVLR is
+embedded in Rust. It is compiled by the Rust compiler and has simple operational
+semantics.
 
-* What is CVLR?
-* How to use CVLR?
-* Overview of all important macros (assert, assume, cex_print...)
\ No newline at end of file
+Assertions
+----------
+
+The simplest feature of CVLR are assertions. An assertion is written as
+`cvlr_assert!(cond)`, where `cond` is the condition being asserted.
+The semantics is the same as traditional asserts in Rust -- an assertion is
+violated if there is (a panic-free) execution that reaches `cvlr_assert!(cond)`
+and in the current execution state `cond` is false.
+
+For example, `cvlr_assert!(true)` is never violated, while `cvlr_assert!(false)`
+is always violated when reached.
+
+What makes `cvlr_assert!()` special is that it is verified symbolically by the
+Certora Prover. That is, the Prover returns `Violated` if there is an input and
+an execution of the program that reaches that assertion and violates it.
+
+Often, the intended meaning of an assertion is to not be violated, i.e., to
+hold on all possible executions. However, sometimes it is useful to check
+whether some execution is possible. In that case, the specification intends for
+the assertion to be reachable. For such cases, CVLR provides a satisfy
+assertion, called `cvlr_satisfy!(cond)`. 
+
+The semantics of `cvlr_satisfy!(cond)` is that it is `Violated` when it is
+either not reached, or when every execution that reaches it, violates the
+condition. For example, `cvlr_satisfy!(true)` is violated only if it is never
+reachable (i.e., part of dead code), and `cvlr_satisfy!(false)` is always
+violated.
+
+Assumptions
+-----------
+
+Assumptions provide a way to restrict input to reflect some pre-condition. CVLR
+provides an assumption macro `cvlr_assume!(cond)`. If an execution reaches
+`cvlr_assume!(cond)`, it continues only if `cond` is true in the current program
+state. Otherwise, the execution aborts.
+
+For example, `cvlr_assume!(true)` is a noop, while `cvlr_assume!(false)` blocks
+all executions that reach it.
+
+A typical use of `cvlr_assume!` is to restrict a range of a value beyond the
+restriction afforded by its type. For example, restricting the maximum value of
+a basis point to `10_000`` can be done as follows:
```

Comment:
```suggestion
a basis point to `10_000` can be done as follows:
```
Do we need the `_` here?
For folks not familiar with the term `basis point` - should we instead you refer to the maximum of some value?